### ISSUE TITLE
got "required field not set" error in nullable field when zero value

### ISSUE THREAD

**Issue by s-hashimoto00** [gogo/protobuf#452]
Only non generated code.
I think so will fix with #449 

reproduce:

proto file
```
syntax = "proto2";

package issue449;

import "github.com/gogo/protobuf/gogoproto/gogo.proto";

message Msg {
	required int64 Int64 = 1 [(gogoproto.nullable) = false];
}
```

go file
```go
package issue449

import (
	"testing"

	"github.com/gogo/protobuf/proto"
)

func TestMsgOne(t *testing.T) {
	if _, err := proto.Marshal(&Msg{Int64: 1}); err != nil {
		t.Fatal(err)
	}
}

func TestMsgZero(t *testing.T) {
	if _, err := proto.Marshal(&Msg{Int64: 0}); err != nil {
		t.Fatal(err)
	}
}
```

run test
```
$ go test -v -run TestMsg ./test/issue449/
=== RUN   TestMsgOne
--- PASS: TestMsgOne (0.00s)
=== RUN   TestMsgZero
--- FAIL: TestMsgZero (0.00s)
        issue449_test.go:17: proto: required field "Int64" not set
FAIL
FAIL    github.com/gogo/protobuf/test/issue449  0.002s
```

**Comment by awalterschulze** [comment]
Thank you for reporting and attempting to fix, we really appreciate it :)

**Comment by donaldgraham** [comment]
@s-hashimoto00, I've merged your pull request.  Can we close this issue now?

**Comment by s-hashimoto00** [comment]
Yes, fixed via #453 
Thank you for review and marge.
